error.c

```diff
@@ -237,7 +234,6 @@ static	void	build_logfile_path(char *buf, const int buf_len)
 void	_dmalloc_open_log(void)
 {
   char	log_path[1024];
```

Comment:
So log_path is 1024 but the internal dprintf buffer is 256.   That's not good.

Reply:
I will provide the new patch to change all hardcode path length(e.g. 1024) to PATH_MAX which is normally 256.

Reply:
PATH_MAX is 4096 on my linux box.  I'd rather just be explicit.  Let's fix the ! and I'll take your change and then do some tweaking of sizes universally.

Reply:
Really appreciate all of your work Xiao!

Reply:
On Linux or other MMU based OS, the stack size is useless since OS will extend the stack as needed, but it is very critical for RTOS, I would suggest to use MAX_PATH as the stack buffer, so:

1. On Linux, it could be 1024 or more big, but who care it:)
2. On RTOS, MAX_PATH is normally defined to small value



> PATH_MAX is 4096 on my linux box. I'd rather just be explicit. Let's fix the ! and I'll take your change and then do some tweaking of sizes universally.

See my reply, we need ! to avoid the warning.